### PR TITLE
fix(stardust): support class-pyramid in theme

### PR DESCRIPTION
**Pull request by T-Wizard**
## Motivation

implements the // TODO - generate class-pyramid
matching what is done in the sense client

### CODE REVIEW

**Review by miralemd — CHANGES_REQUESTED** [review]
We're already using d3-color and should do so for this feature as well to avoid using a custom implementation.

File: apis/theme/src/colors/color-scale.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import Color from './color';
+
+const NaNColor = new Color(0xd2d2d2);
```
Comment (miralemd) [inline apis/theme/src/colors/color-scale.js:3]: Shouldn't the NaN color come from the applied theme?

Comment (T-Wizard) [reply apis/theme/src/colors/color-scale.js]: This match how it is done in the sense client.
I don't think it is important as it should not be used if the scale is correctly defined.
(If it should be from the theme what color in the theme should it use?)

**Comment by T-Wizard** [comment]
now update to use d3-color

File: apis/theme/src/colors/color-scale.js (hunk 1 above)
Comment (miralemd) [reply apis/theme/src/colors/color-scale.js]: Would be nice if we didn't just copy the mistakes from previous code ;) - NaN/Null color is specified in the [theme](https://github.com/qlik-oss/nebula.js/blob/master/apis/theme/src/themes/base.json#L9)

Comment (T-Wizard) [reply apis/theme/src/colors/color-scale.js]: done.
Is it relevant to have a warning when the nan color is used to indicate that there is a problem with the theme?

Comment (miralemd) [reply apis/theme/src/colors/color-scale.js]: No need for a warning specific for the NaN color right now, but if we see a lot of issues of poorly formatted themes then having a theme validator would be nice to have in the future.

**Comment by T-Wizard** [comment]
ping @miralemd: I have fixed all you comments (did I miss something?)

**Comment by miralemd** [comment]
I'm looking at how this affects the theme API input and output - If I understand this correctly, [these lines](https://github.com/qlik-oss/nebula.js/pull/508/files#diff-64d54da90306e9b4fae10be2a7cd06ddR88-R94) mean that there will never be a `'class'` output in `dataScales`, which then also means you can never provide a `'class'` scale that isn't changed internally? So the result in the API output should be to remove `'class'` [here](https://github.com/qlik-oss/nebula.js/blob/master/apis/theme/src/palette-resolver.js#L4) and replace it with `'class-pyramid'`?

**Comment by T-Wizard** [comment]
Yes I think you are right and 'class' should be replaced with 'class-pyramid' and the type of the color property need to changed from {string[]} colors to {string[]|Array<Array<string>>}

**Review by Caele — APPROVED** [review]

**Pull request merged**